src/App.tsx

```diff
@@ -15,36 +19,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'Name',
+  length = 'Length',
+}
+
+interface SortProperty {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPrepareGoods(
+  goods: Goods,
+  { sortField, isReversed }: SortProperty,
+): Goods {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField.includes(SortType.name)) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField.includes(SortType.length)) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPrepareGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  function reset() {
+    setSortField('');
+    setIsReversed(false);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.name)}
+          type="button"
+          className={cn(['button', 'is-info'], {
+            'is-light': !sortField.includes(SortType.name),
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.length)}
+          type="button"
+          className={cn(['button', 'is-success'], {
+            'is-light': !sortField.includes(SortType.length),
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={cn(['button', 'is-warning'], {
+            'is-light': !isReversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || isReversed ? (
```

Comment:
```suggestion
        {sortField || isReversed && (
```
in case you don't have "second" variant to render use this operator, read: https://react.dev/learn/conditional-rendering